Gems/ROS2/Code/Source/Transform/ROS2FrameComponent.cpp

```diff
@@ -0,0 +1,97 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+#include "ROS2/ROS2Bus.h"
+#include "Utilities/ROS2Names.h"
+#include <AzCore/Component/Entity.h>
+#include <AzCore/Serialization/SerializeContext.h>
+
+namespace ROS2
+{
+    void ROS2FrameComponent::Activate()
+    {
+        m_transformPublisher = AZStd::make_unique<TransformPublisher>(GetParentFrameID(), GetFrameID(), GetFrameTransform());
+    }
+
+    void ROS2FrameComponent::Deactivate()
+    {
+        m_transformPublisher.reset();
+    }
+
+    const ROS2FrameComponent* ROS2FrameComponent::GetParentROS2FrameComponent() const
+    {
+        if (const AZ::EntityId parentEntityId = GetEntityTransformInterface()->GetParentId(); parentEntityId.IsValid())
+        {
+            // TODO - is there really no better way???
+            AZ::Entity* parentEntity = nullptr;
+            AZ::ComponentApplicationBus::BroadcastResult(parentEntity, &AZ::ComponentApplicationRequests::FindEntity, parentEntityId);
+            return parentEntity->FindComponent<ROS2FrameComponent>();
+        }
+        return nullptr;
+    }
+
+    const AZ::Transform& ROS2FrameComponent::GetFrameTransform() const
+    {
+        auto transformInterface = GetEntityTransformInterface();
+        if (GetParentROS2FrameComponent() != nullptr)
+        {
+            return transformInterface->GetLocalTM();
+        }
+        return transformInterface->GetWorldTM();
+    }
+
+    AZ::TransformInterface* ROS2FrameComponent::GetEntityTransformInterface() const
+    {
+        return GetEntity()->FindComponent<AzFramework::TransformComponent>();
+    }
+
+    AZStd::string ROS2FrameComponent::GetParentFrameID() const
+    {
+        if (auto parentFrame = GetParentROS2FrameComponent(); parentFrame != nullptr)
+        {
+            return parentFrame->GetFrameID();
+        }
+        return "world";
```

Comment:
This should be parametrized. From my experience - the `map` was used way more frequently than `world`.

Reply:
Added a TODO - this would be a part of a global configuration (which we don't have yet), not a part of each Frame component